pkg/nodejs/pnpm/parse.go

```diff
@@ -95,15 +116,61 @@ func (p *Parser) parse(lockFile *LockFile) ([]types.Library, []types.Dependency)
 	return libs, deps
 }
 
-func isIndirectLib(name string, directDeps map[string]string) bool {
+func isIndirectLib(name string, directDeps map[string]interface{}) bool {
 	_, ok := directDeps[name]
 	return !ok
 }
 
-func getPackageNameAndVersion(pkg string) (string, string) {
-	idx := strings.LastIndex(pkg, "/")
-	name := pkg[1:idx]
-	version := pkg[idx+1:]
-
+func getPackageNameAndVersion(pkg, name, version string, lockFileVersion float64) (string, string) {
```

Comment:
We should use the same terminology in pnpm.

```suggestion
func getPackageNameAndVersion(depPath, name, version string, lockFileVersion float64) (string, string) {
```

Reply:
Made in https://github.com/aquasecurity/go-dep-parser/pull/220/commits/627bec841fb40712e7e842380ef56b3153929377